mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/util/generation/GeneratedCodePatterns.java

```diff
@@ -259,6 +292,113 @@ public static void setValue(
     }
   }
 
+  private static void generateParameterizedSet(
+      CodeBlock cqlName,
+      CodeBlock valueExtractor,
+      String targetName,
+      MethodSpec.Builder methodBuilder,
+      TypeName typeName,
+      BindableHandlingSharedCode enclosingClass,
+      CodeBlock nullSavingStrategy) {
+    generateSetWithNullSavingStrategy(
+        cqlName,
+        valueExtractor,
+        targetName,
+        methodBuilder,
+        nullSavingStrategy,
+        CodeBlock.of(
+            "$1L = $1L.set($2L, $3L, $4L)",
+            targetName,
+            cqlName,
+            valueExtractor,
+            enclosingClass.addGenericTypeConstant(typeName)));
+  }
+
+  private static void generateSetWithClass(
+      CodeBlock cqlName,
+      CodeBlock valueExtractor,
+      String targetName,
+      MethodSpec.Builder methodBuilder,
+      TypeName typeName,
+      CodeBlock nullSavingStrategy) {
+
+    generateSetWithNullSavingStrategy(
+        cqlName,
+        valueExtractor,
+        targetName,
+        methodBuilder,
+        nullSavingStrategy,
+        CodeBlock.of(
+            "$1L = $1L.set($2L, $3L, $4T.class)", targetName, cqlName, valueExtractor, typeName));
+  }
+
+  private static void generatePrimitiveSet(
+      CodeBlock cqlName,
+      CodeBlock valueExtractor,
+      String targetName,
+      MethodSpec.Builder methodBuilder,
+      String primitiveAccessor,
+      CodeBlock nullSavingStrategy) {
+
+    // todo how to handle primitive type getters? (int id == null is not possible)
```

Comment:
@olim7t should we handle primitive types?
Primitive types cannot have null value so I think null handling strategy does not apply

Reply:
that sounds right to me, no special logic should be needed for primitives :+1:

Reply:
Agreed. If you want the ability to do partial updates on a property, it should be mapped to the boxed type.